server/src/main/java/io/crate/expression/tablefunctions/FoobarFunction.java

```diff
@@ -0,0 +1,121 @@
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.tablefunctions;
+
+import io.crate.data.Input;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.Signature;
+import io.crate.metadata.tablefunctions.TableFunctionImplementation;
+import io.crate.types.DataTypes;
+import io.crate.types.RowType;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public class FoobarFunction extends TableFunctionImplementation<List<Object>> {
+
+    public static final String NAME = "foobar";
+    private static final RowType ROW_TYPE = new RowType(
+        List.of(DataTypes.STRING_ARRAY), List.of("planetzs")); // single column, array(text)
+
+    public static void register(TableFunctionModule module) {
+        module.register(
+            Signature.table(NAME, ROW_TYPE.getTypeSignature()),
+            (signature, args) -> new FoobarFunction(
+                signature,
+                new FunctionInfo(
+                    new FunctionIdent(NAME, args),
+                    ROW_TYPE,
+                    FunctionInfo.Type.TABLE))
+        );
+    }
+
+    private final Signature signature;
+    private final FunctionInfo info;
+
+    private FoobarFunction(Signature signature, FunctionInfo info) {
+        this.signature = signature;
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Signature signature() {
+        return signature;
+    }
+
+    @Override
+    public RowType returnType() {
+        return ROW_TYPE;
+    }
+
+    @Override
+    public boolean hasLazyResultSet() {
+        return false;
+    }
+
+    @Override
+    public Scalar<Iterable<Row>, List<Object>> compile(List<Symbol> arguments) {
+        return this;
+    }
+
+    @Override
+    public Iterable<Row> evaluate(TransactionContext txnCtx, Input[] args) {
+        return iterableOf(List.of(
+            "Mercury", "Venus", "Earth", "Jupiter", "Saturn", "Your-anus", "Ne pais fume"));
+    }
+    private static Iterable<Row> iterableOf(List<String> data) {
+        if (data == null || data.size() == 0) {
+            return null;
+        }
+        return () -> new Iterator<>() {
+            final RowN row = new RowN(1);
+            int idx = 0;
+
+            @Override
+            public boolean hasNext() {
+                return idx < 1;
+            }
+
+            @Override
+            public Row next() {
+                if (!hasNext()) {
+                    throw new NoSuchElementException("no more rows");
+                }
+                row.cells(data.toArray());
```

Comment:
`cells` is a varargs argument, this will cause you to have a Row with multiple cells.


```
 List.of("foo", "bar").toArray()
$2 ==> Object[2] { "foo", "bar" }


var row = new RowN(List.of("foo", "bar").toArray())
row.numColumns(); -> this will be 2, instead of 1
```


What you want is something like this:


```
var row = new RowN(new Object[] { List.of("foo", "bar").toArray() });
row.numColumns(); -> this will be 2, instead of 1
```

Reply:
and then the problem is:

```
SQLActionException[ClassCastException: class [Ljava.lang.Object; cannot be cast to class java.util.List ([Ljava.lang.Object; and java.util.List are in module java.base of loader 'bootstrap')]
```

Reply:
cool

Reply:
YES!!!!!!! thank you very much!!!! I could not see the freaking varargs issue!!! and went mental, hitting my antlers against the wall, xxx

Reply:
> and then the problem is:

Right, the `toArray()` call is bogus.  `STRING_ARRAY` value must be a `List<String>`, not an array.